homeassistant/components/shelly/binary_sensor.py

```diff
@@ -272,6 +272,26 @@ def __init__(
         entity_category=EntityCategory.DIAGNOSTIC,
         entity_class=RpcBluTrvBinarySensor,
     ),
+    "flood": RpcBinarySensorDescription(
+        key="flood",
+        sub_key="alarm",
+        name="Flood",
+        device_class=BinarySensorDeviceClass.MOISTURE,
+    ),
+    "mute": RpcBinarySensorDescription(
+        key="flood",
+        sub_key="mute",
+        name="Mute",
+        device_class=BinarySensorDeviceClass.SOUND,
+    ),
+    "error": RpcBinarySensorDescription(
+        key="flood",
+        sub_key="errors",
+        name="Error",
+        device_class=BinarySensorDeviceClass.PROBLEM,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        extra_state_attributes=lambda value: {"error": value},
```

Comment:
I think we should avoid expanding the usage of `extra_state_attributes`, currently we have only 3 usages of `extra_state_attributes`, all in Gen1 which added in very early stages of the integration. 
If this information is needed we should create another entity with it.

Reply:
We can go with either approaches, but for me the error info is so related to the binary sensor status that makes more sense to have then in a single "place".

Reply:
If we know that the `errors` value can contain a list of errors then we should implement a separate entity for each type of error.

Reply:
The documentation mentions a `cable_unplugged` error:

![obraz](https://github.com/user-attachments/assets/96cfec8a-fbb9-44d2-8e36-ccf8793d7fe5)